### DISCUSSION TITLE
How to display a structured database?

### DISCUSSION THREAD

**Discussion by jS5t3r** [alshedivat/al-folio#2770] [Q&A] [answered]
How to display a structured database with 100 entries and more? It has for example 3 columns. 
This database will be automatically updated from time to time.

**Comment by riasat-sheikh** [comment]
Maybe you can write a script that can parse the data and then render that in your website.

**Comment by george-gca** [comment]
> This database will be automatically updated from time to time.

Can you be more specific? If this time to time is long enough you can have different solutions than one with more frequent updates.

**Comment by jS5t3r** [comment]
@george-gca sure. 
Final goal: https://nicholas.carlini.com/writing/2019/all-adversarial-example-papers.html

Current status: 
I have a GitHub workflow which show update papers once per day: https://github.com/lorenz-peter/lorenz-peter.github.io/actions/runs/11356739477/job/31588462669 (it basically executes this script: https://github.com/lorenz-peter/lorenz-peter.github.io/blob/master/papers/fetch_papers.py) 

Actually, it ought to should create a json file like this one: 
https://github.com/lorenz-peter/lorenz-peter.github.io/blob/master/papers/papers.json
Q: I dont know how read the papers.json from the the workflow. How do this? 

This json file should be read out and and shown on my website. Maybe as post https://github.com/lorenz-peter/lorenz-peter.github.io/tree/master/_posts or project https://github.com/lorenz-peter/lorenz-peter.github.io/tree/master/_projects
Q: Best way to show the json file as table online?

**Reply by george-gca** [reply, marked as answer]
I believe using Bootstrap table will fit your needs. Check the last example in our [sample post](https://alshedivat.github.io/al-folio/blog/2023/tables/). It basically reads the content from a json file.

**Reply by jS5t3r** [reply]
@george-gca  This is great. 

The only open think is that how to update automatically this table then.

**Reply by george-gca** [reply]
To update the table automatically you'll have to trigger the build of the whole page everyday (or every X days). For that you can create a new GitHub action that would be a merge of [.github/workflows/deploy.yml](https://github.com/alshedivat/al-folio/blob/master/.github/workflows/deploy.yml) and [.github/workflows/schedule-posts.txt](https://github.com/alshedivat/al-folio/blob/master/.github/workflows/schedule-posts.txt). Check our [CUSTOMIZE.md](https://github.com/alshedivat/al-folio/blob/master/CUSTOMIZE.md#scheduled-posts) about the scheduled posts.

**Reply by jS5t3r** [reply]
@george-gca  Thank you for your help.  I have closed this dicsussion.
